gremlin-go/driver/graphTraversal.go

```diff
@@ -669,6 +670,30 @@ func (g *GraphTraversal) Write(args ...interface{}) *GraphTraversal {
 	return g
 }
 
+func int32Args(args ...interface{}) []interface{} {
+	for i, arg := range args {
+		switch val := arg.(type) {
+		case uint8:
+			args[i] = int32(val)
+		case uint32:
+			args[i] = int32(val)
+		case uint64:
```

Comment:
Should we include `uint64` and `int64`? Just in case someone enters a number larger than int32, casting would not give the correct result. I wonder if we should just propagates those types through, and let server error?

Reply:
Then we might need to exclude also `int` and `uint`, right? I feel that would make this patch not very useful. If I'm not wrong, the common use case `.Times(123)` would not be covered anymore.

From the Go's spec:

```
There is also a set of predeclared integer types with implementation-specific sizes:

uint     either 32 or 64 bits
int      same size as uint
```

Reply:
Yeah, main reason we are running into the issue is we needed to serialize go int into java long since it could be 64 bits. I'm more in favour of leaving letting the error be returned by the server instead of casting, and perhaps adding documentations so users know what to do in those case.

Reply:
Also, the reason to suggest excluding `uint64` and `int64` is because I think we should enforce the fact that these steps only takes 32-bit integers or less, so we shouldn't allow users to consciously input types that is larger. 

The ambiguity comes with `int`, as that is the default integer type for go if one doesn't specify the type, so that would be the main case we need to handle. 

I'm not entirely sure how `uint` would be handled, since a 32-bit `uint` would technically fit to a long in java. Perhaps if we can figured out a good way to cast, this can be handled a similar way.